tests/php/ORM/DataObjectTest.php

```diff
@@ -1915,6 +1915,24 @@ public function testFieldTypes()
         $this->assertEquals('1988-03-04 06:30:00', $obj->DatetimeField);
     }
 
+    public function testZeroIsFalse()
+    {
+        $obj = new DataObjectTest\Fixture();
+        $obj->MyInt = 0;
+        $obj->MyDecimal = 0.00;
+        $obj->MyCurrency = 0.00;
+        $obj->write();
+
+        $this->assertFalse((bool)$obj->MyInt);
+        $this->assertFalse((bool)$obj->MyDecimal);
+        $this->assertFalse((bool)$obj->MyCurrency);
+
+        $obj2 = DataObjectTest\Fixture::get()->byId($obj->ID);
+        $this->assertFalse((bool)$obj2->MyInt); // return false as hoped
+        $this->assertFalse((bool)$obj2->MyDecimal); // returns true, bug
+        $this->assertFalse((bool)$obj2->MyCurrency); // returned true, bug
```

Comment:
Tests are failing. Would you mind tossing in some comments on each of these `assertFalse` so it's more obvious what behaviour is failing?